crates/core/src/db/datastore/locking_tx_datastore/btree_index.rs

```diff
@@ -62,14 +62,26 @@ impl Iterator for BTreeIndexIter<'_> {
 /// [BTreeIndex]
 pub struct BTreeIndexRangeIter<'a> {
     range_iter: btree_set::Range<'a, IndexKey>,
+    keys_scanned: u64,
```

Comment:
Nit: I would do `num_keys_scanned` or `keys_scanned_count`, so that it's clear that it's a count and not some identifier.